ui/raidboss/emulator/EmulatorCommon.ts

```diff
@@ -0,0 +1,127 @@
+import { LocaleNetRegex } from '../../../resources/translations';
+import NetRegexes from '../../../resources/netregexes';
+
+// Disable ban-types for cloneData as it needs to work on raw objects for performance reasons.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type DataType = { [key: string]: any } | null;
+
+export default class EmulatorCommon {
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  static cloneData(data: DataType, exclude = ['options', 'party']): DataType {
+    let ret: DataType;
+    if (Array.isArray(data))
+      ret = [];
+    else
+      ret = {};
+
+    // Use extra logic for top-level extend for property exclusion
+    // This cut the execution time of this code from 41,000ms to 50ms when parsing a 12 minute pull
+    for (const i in data) {
+      if (exclude.includes(i))
+        continue;
+
+      if (typeof data[i] === 'object')
+        ret[i] = EmulatorCommon._cloneData(data[i]);
+      else
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```

Comment:
```suggestion
        // Assignment of any to any.  See DataType definition above for reasoning.
        // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
```

I think any eslint-disable should at least get some explanation as to why, even if it's just pointing to something else.

Reply:
Agreed. The multiple comments above an un-braced `else` strikes me as a bit painful, though. Added comment for next commit.